conflictlog/record.go

```diff
@@ -94,11 +84,73 @@ func (x *Xattrs) String() string {
 // ConflictRecord has the all the details of the detected conflict
 // which are needed to be persisted
 type ConflictRecord struct {
+	Timestamp     string `json:"timestamp"`
 	Id            string `json:"id"`
 	DocId         string `json:"docId"`
 	ReplicationId string `json:"replId"`
 	Source        DocInfo
 	Target        DocInfo
+
+	// Note: The following will not be serialized to json
+	datatype uint8
+	body     []byte
+}
+
+// Id and ReplicationId are derived/generated during conflict logging.
+func NewConflictRecord(key string, srcScopeName, tgtScopeName, srcCollectionName, tgtCollectionName, srcBucketUUID, tgtBucketUUID, srcClusterUUID, tgtClusterUUID, srcHostname, tgtHostname string,
```

Comment:
We have legacy XDCR code that has a constructor with a long list of inputs.
Is it possible to move away from that here for readability sake?

Reply:
It so happens that me & @sumukhbhat2701 discussed this. I agree this can become unwieldy. Noted.

Reply:
Will remove the constructor.